internal/istioctl/istioctl.go

```diff
@@ -286,3 +279,22 @@ func fetchIstioctl(homeDir string, targetDistribution *manifest.IstioDistributio
 	}
 	return nil
 }
+
+func fetchIstioctlURL(targetDistribution *manifest.IstioDistribution, runtimeGOOS string, runtimeGOARCH string) string {
+	const (
+		istioctlDownloadURLFormatWithArch    = "https://istio.tetratelabs.io/getmesh/files/istio-%s-%s-%s.tar.gz"
+		istioctlDownloadURLFormatWithoutArch = "https://istio.tetratelabs.io/getmesh/files/istio-%s-%s.tar.gz"
+	)
+
+	var url string
+	if runtimeGOOS != "darwin" {
+		url = fmt.Sprintf(istioctlDownloadURLFormatWithArch, targetDistribution.String(), runtimeGOOS, runtimeGOARCH)
+		return url
+	}
+	if runtimeGOARCH == "arm64" {
+		url = fmt.Sprintf(istioctlDownloadURLFormatWithArch, targetDistribution.String(), "osx", runtimeGOARCH)
+	} else {
+		url = fmt.Sprintf(istioctlDownloadURLFormatWithoutArch, targetDistribution.String(), "osx")
+	}
+	return url
```

Comment:
Maybe we could remove "url" out 👍 
```suggestion
	if runtimeGOOS != "darwin" {
		return fmt.Sprintf(istioctlDownloadURLFormatWithArch, targetDistribution.String(), runtimeGOOS, runtimeGOARCH)
	}

	if runtimeGOARCH == "arm64" {
		return fmt.Sprintf(istioctlDownloadURLFormatWithArch, targetDistribution.String(), "osx", runtimeGOARCH)
	} else {
		return fmt.Sprintf(istioctlDownloadURLFormatWithoutArch, targetDistribution.String(), "osx")
	}
```